src/app/clusters/ota-server/ota-server.cpp

```diff
@@ -28,26 +28,52 @@
 
 #include <support/Span.h>
 
+#include "ota-server-delegate.h"
+
 using namespace chip;
+using namespace chip::app::clusters;
 
-// TODO: all callbacks need implementations
+namespace {
+constexpr uint8_t kLocationParamLength   = 2;   // The expected length of the Location parameter in QueryImage
+constexpr size_t kMaxMetadataLen         = 512; // The maximum length of Metadata in any OTA Server command
+constexpr size_t kUpdateTokenParamLength = 32;  // The expected length of the Update Token parameter used in multiple commands
+} // namespace
 
 /**
  * @brief OTA Software Update Server Cluster ApplyUpdateRequest Command callback
  *
+ * @note It is the application's reponsibility to send the ApplyUpdateRequestResponse command after this is handled.
+ *
  * @param updateToken Identifier for the Software Image to be applied. Should be 32 octets long.
  * @param newVersion The SoftwareVersion value of the new Software Image that the client is ready to apply.
  */
 
 bool emberAfOtaSoftwareUpdateServerClusterApplyUpdateRequestCallback(app::Command * commandObj, chip::ByteSpan updateToken,
                                                                      uint32_t newVersion)
 {
-    return false;
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    ChipLogDetail(Zcl, "OTA Server received ApplyUpdateRequest");
+
+    if (updateToken.size() != kUpdateTokenParamLength)
+    {
+        // TODO: correct error?
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_FIELD);
```

Comment:
Is `EMBER_ZCL_STATUS_INVALID_FIELD` the correct error to return for when a parameter is unexpected length?

Reply:
@vivien-apple ?

Reply:
CHIP spec has a new status code:
```
| INVALID_ARGUMENT        | 0xc6  | Argument(s) specified in a command are invalid for that particular command invocation
```

It is possible it is not defined yet. You could check.

Reply:
Got it, I defined a new error and I'll set it to 0xC6